erpnext/controllers/buying_controller.py

```diff
@@ -420,4 +420,39 @@ def validate_schedule_date(self):
 				if d.schedule_date and getdate(d.schedule_date) < getdate(self.transaction_date):
 					frappe.throw(_("Expected Date cannot be before Transaction Date"))
 		else:
-			frappe.throw(_("Please enter Schedule Date"))
\ No newline at end of file
+			frappe.throw(_("Please enter Schedule Date"))
+
+	def apply_shipping_rule(self):
```

Comment:
Please be a DRY programmer

https://github.com/frappe/erpnext/blob/develop/erpnext/controllers/selling_controller.py#L70

Reply:
Move this to a common method in `shipping_rule.py`

Reply:
@rmehta Done. But I have put the common code in transaction.js and accounts_controller.py